File: HotelBookingAutomation/HotelBookingTests/PageObjectModel/WebManager.cs
using HotelBookingTests.PageObjectModel.Pages;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HotelBookingTests.PageObjectModel
{
    class WebManager
    {
        private static IWebDriver Driver { get; set; }

        public static void Initialize()
        {
            if (Driver == null)
            {
                Driver = GetWebdriver(Config.BrowserType, Config.ImplicitWaitTimeOutInMilliseconds, false, Config.DriverLocation);
                CreatePages();
            }
        }

        private static IWebDriver GetWebdriver(string browserType,
                                                int implicitWaitTimeOutInMilliseconds,
                                                bool openInPrivateMode,
                                                string driverLocation = "")
        {
            string driverDirectoryPath = driverLocation;
            if (string.IsNullOrEmpty(driverDirectoryPath))
            {
                driverDirectoryPath = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
            }

            int timeoutInMinutes = 30;

            IWebDriver iWebdriver = null;
            switch (browserType)
            {
                case "chrome":
                    {
                        ChromeOptions chromeOptions = new ChromeOptions();
                        chromeOptions.AddArgument("no-sandbox");
                        chromeOptions.AddArguments("disable-infobars");
                        if (openInPrivateMode)
                        {
                            chromeOptions.AddArguments("incognito");
                        }

                        iWebdriver = new ChromeDriver(driverDirectoryPath, chromeOptions, TimeSpan.FromMinutes(timeoutInMinutes));

                        iWebdriver.Manage().Window.Maximize();
                    }
                    break;

                default:
                    throw new Exception($"Testing on \"{browserType}\" browser is not yet supported.");
            }

            iWebdriver.Manage().Timeouts().ImplicitWait = TimeSpan.FromMilliseconds(implicitWaitTimeOutInMilliseconds);
            return iWebdriver;
        }

        public static void Cleanup()
        {
            Driver.Quit();
        }

        public static void OpenWebpage(string strURL)
        {
            Driver.Navigate().GoToUrl(strURL);
        }

        public static string URL { get { return Driver.Url; } }

        private static void CreatePages()
        {
            HotelBookingPage = new HotelBookingPage(Driver);
        }

        public static IHotelBookingPage HotelBookingPage { get; private set; }
    }
}

File: HotelBookingAutomationJava/src/test/java/pageObjectModel/WebManager.java
package pageObjectModel;

import pageObjectModel.pages.HotelBookingPage;
import pageObjectModel.pages.HotelBookingPageImpl;
import org.apache.commons.lang3.StringUtils;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.TimeUnit;

public class WebManager {
    private WebDriver driver;

    public WebDriver getDriver() {
        return driver;
    }

    public void initialize() throws Exception {
        if (driver == null) {
            driver = getWebdriver("chrome", 10000, false, "");
            createPages();
        }
    }

    private WebDriver getWebdriver(String browserType,
                                   int implicitWaitTimeOutInMilliseconds,
                                   boolean openInPrivateMode,
                                   String driverLocation) throws Exception {
        String driverDirectoryPath = driverLocation;
        if (StringUtils.isBlank(driverDirectoryPath)) {
            driverDirectoryPath = System.getProperty("user.dir") + "/drivers/chromedriver.exe";
        }

        System.setProperty("webdriver.chrome.driver", driverDirectoryPath);

        int timeoutInMinutes = 30;

        WebDriver webdriver = null;
        switch (browserType) {
            case "chrome": {
                ChromeOptions chromeOptions = new ChromeOptions();
                List<String> arguments = new ArrayList<String>();
                arguments.add("no-sandbox");
                arguments.add("disable-infobars");
                if (openInPrivateMode) {
                    arguments.add("incognito");
                }
                chromeOptions.addArguments(arguments);

                webdriver = new ChromeDriver(chromeOptions);

                webdriver.manage().window().maximize();
            }
            break;

            default:
                throw new Exception("Testing on \"" + browserType + "\" browser is not yet supported.");
        }

        webdriver.manage().timeouts().implicitlyWait(implicitWaitTimeOutInMilliseconds, TimeUnit.MILLISECONDS);
        return webdriver;
    }

    public void cleanup() {
        driver.manage().deleteAllCookies();
        driver.quit();
    }

    public void openWebpage(String strURL) {
        driver.navigate().to(strURL);
    }

    public String getURL() {
        return driver.getCurrentUrl();
    }

    private void createPages() {
        hotelBookingPage = new HotelBookingPageImpl(driver);
    }

    private HotelBookingPage hotelBookingPage;

    public HotelBookingPage getHotelBookingPage() {
        return hotelBookingPage;
    }
}
